engine/lite/embeddedDeploymentManager/src/test/scala/pl/touk/nussknacker/streaming/embedded/RequestResponseEmbeddedDeploymentManagerTest.scala

```diff
@@ -0,0 +1,85 @@
+package pl.touk.nussknacker.streaming.embedded
+
+import akka.actor.ActorSystem
+import com.typesafe.config.ConfigFactory
+import com.typesafe.config.ConfigValueFactory.fromAnyRef
+import org.scalatest.{FunSuite, Matchers}
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.deployment.simple.SimpleStateStatus
+import pl.touk.nussknacker.engine.api.deployment.{DeployedScenarioData, DeploymentManager, ProcessingTypeDeploymentServiceStub}
+import pl.touk.nussknacker.engine.api.process.{EmptyProcessConfigCreator, ProcessName}
+import pl.touk.nussknacker.engine.build.RequestResponseScenarioBuilder
+import pl.touk.nussknacker.engine.deployment.{DeploymentData, User}
+import pl.touk.nussknacker.engine.embedded.RequestResponseEmbeddedDeploymentManagerProvider
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.spel.Implicits._
+import pl.touk.nussknacker.engine.testing.LocalModelData
+import pl.touk.nussknacker.test.{AvailablePortFinder, PatientScalaFutures}
+import sttp.client.{HttpURLConnectionBackend, Identity, NothingT, SttpBackend, UriContext, basicRequest}
+import sttp.model.StatusCode
+
+import scala.concurrent.Future
+
+class RequestResponseEmbeddedDeploymentManagerTest extends FunSuite with Matchers with PatientScalaFutures {
+
+  private implicit val backend: SttpBackend[Identity, Nothing, NothingT] = HttpURLConnectionBackend()
+
+  protected def prepareFixture(initiallyDeployedScenarios: List[DeployedScenarioData] = List.empty): FixtureParam = {
+
+    val modelData = LocalModelData(ConfigFactory.empty().withValue("components.kafka.disabled", fromAnyRef(true)), new EmptyProcessConfigCreator)
+    implicit val deploymentService: ProcessingTypeDeploymentServiceStub = new ProcessingTypeDeploymentServiceStub(initiallyDeployedScenarios)
+    implicit val as: ActorSystem = ActorSystem("test")
+    implicit val dummyBackend: SttpBackend[Future, Nothing, NothingT] = null
+    import as.dispatcher
```

Comment:
if we do it here then why do we need to do it inside RequestResponseDeploymentStrategy ?

Reply:
hmmm... maybe DeploymentManagerProvider needs rethinking... for now I changed RequestResponseDeploymentStrategy